dns/pfSense-pkg-bind9/files/usr/local/pkg/bind.inc

```diff
@@ -513,7 +510,10 @@ EOD;
 							}
 						}
 						for ($y = 0; $y < sizeof($zone['row']); $y++) {
-							$hostname = (preg_match("/(MX|NS)/", $zone['row'][$y]['hosttype']) ? "@" : $zone['row'][$y]['hostname']);
+							$hostname = $zone['row'][$y]['hostname'];
+							if (preg_match("/(MX|NS)/", $zone['row'][$y]['hosttype']) && trim($hostname) == false) {
```

Comment:
The old code checked if the hosttype was MX or NS, then it would always substitute "@" for the hostname. The trim($hostname) check is so that this substitution only occurs if hostname was left empty, which makes this change backward compatible. Otherwise, the hostname provided by the user is kept.